File: Go/01.Go-Basic-February-2022/09.advancedLoops-Lab/05.accountBalance.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	var bankAccount float32 = 0

	var input string
	fmt.Scanln(&input)

	for input != "NoMoreMoney" {
		sum, _ := strconv.ParseFloat(input, 32)

		if sum >= 0 {
			bankAccount += float32(sum)
			fmt.Printf("Increase: %.2f\n", sum)
		} else {
			fmt.Println("Invalid operation!")
			break
		}

		fmt.Scanln(&input)
	}

	fmt.Printf("Total: %.2f", bankAccount)
}

File: JavaScript/01.JavaScript-Basic-February-2021/09.whileLoop-Lab/05.accountBalance.js
function accountBalance(input) {
    let sum = 0;
    let index = 0;
    let current = input[index];

    while (current != "NoMoreMoney") {
        let num = Number(current);

        if (num < 0) {
            console.log("Invalid operation!");
            break;
        } else {
            sum += num;
            console.log(`Increase: ${num.toFixed(2)}`);
        }
        index++;
        current = input[index];
    }
    console.log(`Total: ${sum.toFixed(2)}`);
}

accountBalance(["5.51", "69.42", "100", "NoMoreMoney"]);
accountBalance(["120", "45.55", "-150"]);